includes/fbproduct.php

```diff
@@ -631,6 +681,61 @@ public function prepare_product( $retailer_id = null, $prepare_for_product_feed
 			);
 		}
 
+		public function get_fb_enhanced_attribute($attribute_key) {
+			$attribute_key = strtolower($attribute_key); // Make it easier to match
+
+			$attr_value = get_post_meta(
+				$this->id,
+				$this->build_enhanced_attribute_meta_key($attribute_key),
+				true
+			);
+
+			if(!is_null($attr_value) && strlen($attr_value) > 0) {
+				return $attr_value;
+			}
+			$category_handler = facebook_for_woocommerce()->get_facebook_category_handler();
+			foreach($this->get_attributes() as $attribute) {
+				$name = strtolower($attribute->get_name());
+				$value = wc_implode_text_attributes($attribute->get_options());
+				if($name === $attribute_key && $category_handler->is_valid_value_for_attribute($attribute_key, $value)){
+					return $value;
+				}
+			}
+			return null;
+		}
+
+		/**
+		* Adds enhanced catalog fields to product data array. Separated from
+		* the main function to make it easier to develop and debug, potentially
+		* worth refactoring into main prepare_product function when complete.
+		*
+		* @param array $product_data map
+		* @return array
+		*/
+		private function apply_enhanced_catalog_fields_from_attributes($product_data) {
+			// So weird but worth noting
+			// name is used on the batch api, the same field is title on the feed
+			// image_url is used on the batch api, the same field is image_link on the feed
+			// url is used on the batch api, the same field is link on the feed
+
+			$fb_category_id = $this->get_fb_category();
+			$category_handler = facebook_for_woocommerce()->get_facebook_category_handler();
+			if(!$fb_category_id || !$category_handler->is_category($fb_category_id)){
+				return $product_data;
+			}
+			$enhanced_data = array();
+			$enhanced_data['fb_product_category'] = $fb_category_id;
```

Comment:
So this will be google product category in a bit, making those changes now but the attribute mapping is more awkward as there are so many more categories.